mlguess/keras/models.py

```diff
@@ -307,22 +309,22 @@ def load_model(cls, conf):
 
     def mae(self, y_true, y_pred):
         """ Compute the MAE """
-        num_splits = y_pred.shape[-1]
-        if num_splits == 4:
-            mu, _, _, _ = tf.split(y_pred, num_splits, axis=-1)
-        elif num_splits == 2:
-            mu, _ = tf.split(y_pred, num_splits, axis=-1)
+        # num_splits = y_pred.shape[-1]
+        if self.num_splits == 4:
```

Comment:
You may be able to do `mu = tf.split(y_pred, num_splits, axis=-1)[0]` or similar to prevent specializations for the different `num_splits` cases.